### PR TITLE
[RFC] vim-patch:7.4.17[83,85,93]

### PR DESCRIPTION
**Pull request by brcolow**
7.4.1783:
Problem:    The old regexp engine doesn't handle character classes correctly.
            (Manuel Ortega)
Solution:   Use regmbc() instead of regc().  Add a test.

https://github.com/vim/vim/commit/af98a49dd0ef1661b4998f118151fddbf6e4df75

7.4.1785:

Problem:    Regexp test fails on windows.
Solution:   set 'isprint' to the right value for testing.

https://github.com/vim/vim/commit/490465bda6ab66f78041709cc02f48a25486a3e5

7.4.1793:

Problem:    Some character classes may differ between systems.  On OS/X the
            regexp test fails.
Solution:   Make this less dependent on the system. (idea by Kazunobu Kuriyama)

https://github.com/vim/vim/commit/e8aee7dcf9b12becff86e8ce1783a86801c5f9f6

Note: Currently the commits are out-of-order...but fixing the order would create conflicts that would be unnecessary work IMHO. If this must be fixed, LMK and I will tackle it.

Currently failing...but the two strings for the "expected" and "got" *appear* to be identical.

### CODE REVIEW

**Comment by Shougo** [comment]
7.4.1783, 7.4.1785 are also needed.

**Review by mhinz — APPROVED** [review]

File: src/nvim/testdir/test_expr_utf8.vim [hunk 1]
```diff
@@ -35,3 +35,60 @@ func Test_strcharpart_utf8()
...
+      let xdigitchars .= c
+    endif
+    let i += 1
+  endwhile
+
+  call assert_equal('ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz', alphachars)
+  call assert_equal('abcdefghijklmnopqrstuvwxyzµßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿ', lowerchars)
+  call assert_equal('ABCDEFGHIJKLMNOPQRSTUVWXYZÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞ', upperchars)
+  call assert_equal('0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz', alnumchars)
+  call assert_equal(' !"#$%&''()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~ ¡¢£¤¥¦§¨©ª«¬­®¯°±²³´µ¶·¸¹º»¼½¾¿ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõö÷øùúûüýþÿ', printchars)
```
Comment (jamessan) [inline src/nvim/testdir/test_expr_utf8.vim]: The character between `~` and `¡` is supposed to be a non-breaking space, not a space.

**Review by jamessan — COMMENTED** [review]
> Currently failing...but the two strings for the "expected" and "got" appear to be identical.

*cough*Copy/paste is good*cough*

**Comment by jamessan** [comment]
This should be good.  Can you rebase on master?

**Comment by brcolow** [comment]
Rebased.

**Pull request merged**